### PR TITLE
Extract column features from AutoMM's model outputs

### PR DESCRIPTION
**Pull request by zhiqiangdon**
1. Add one util function to extract column features for both image and text.
2. Support extracting column features for models `timm_image`, `hf_text`, and `clip`.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1718-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1718/1/index.html

File: text/src/autogluon/text/automm/models/clip.py [hunk 1]
```diff
@@ -98,28 +118,60 @@ def forward(
 
         assert images.dim() == 5
         b, n, c, h, w = images.shape
-        image_features = self.model.get_image_features(
+        vision_outputs = self.model.vision_model(
```
Comment (bryanyzhu) [inline text/src/autogluon/text/automm/models/clip.py:121]: Just a question, why we want to do this naming change, like from image -> vision?

File: text/src/autogluon/text/automm/models/fusion.py [hunk 2]
```diff
@@ -67,6 +67,7 @@ def __init__(
             Basically, it supports adding an auxilliary loss for each individual model.
         """
         super().__init__()
+        logger.debug("initializing MultimodalFusionMLP")
```
Comment (bryanyzhu) [inline text/src/autogluon/text/automm/models/fusion.py:70]: Do we want to leave debug logger here?

**Review by bryanyzhu — COMMENTED** [review]
Nice work. Just left two questions from my side, will approve once I get a better understanding.

File: text/src/autogluon/text/automm/models/clip.py (hunk 1 above)
Comment (zhiqiangdon) [reply text/src/autogluon/text/automm/models/clip.py:121]: Changing from `get_image_features` to `vision_model` is to be consistent with changing from `get_text_features` to `text_model`. See https://github.com/huggingface/transformers/blob/v4.14.1/src/transformers/models/clip/modeling_clip.py#L1006-L1010

File: text/src/autogluon/text/automm/models/fusion.py (hunk 2 above)
Comment (zhiqiangdon) [reply text/src/autogluon/text/automm/models/fusion.py:70]: This is to be consistent with the debug loggers in other models' init functions. The debug messages are helpful to track whether code is run as expected in the debug mode. By default, debug message is not printed unless the `verbosity` hyper-parameter is set to 4.

File: text/src/autogluon/text/automm/models/clip.py (hunk 1 above)
Comment (bryanyzhu) [reply text/src/autogluon/text/automm/models/clip.py:121]: I see, thank you for the reference.

**Review by bryanyzhu — APPROVED** [review]

**Review by sxjscience — APPROVED** [review]

**Pull request merged**